### DISCUSSION TITLE
Is it normal for nextJS to be running getServerSideProps even when navigating through next router?

### DISCUSSION THREAD

**Discussion by serendipity1004** [vercel/next.js#14574] [Help] [answered]
when I create route `/one` and `/two` and have getServerSideProps in each of them, even if I route through clicking on next `Link` I realized that getServerSiedeProps runs. However, I thought that only the original request runs in the server and subsequent navigation happens through client side rendering. If that is the case, getServerSideProps should only fire during the initial request but obviously that doesn't seem to be. Is this because I am currently running in dev mode or is it how it is supposed to be?

**Comment by arso1er** [comment]
Hi,

From what I've read in the doc, it's the expected behaviour. You can read about data fetching in nextjs at [https://nextjs.org/docs/basic-features/data-fetching](https://nextjs.org/docs/basic-features/data-fetching).

**Comment by lfades** [comment, marked as answer]
It is how it is supposed to be, your page will be **server side rendered**, meaning that every request will require `getServerSideProps` to run, if you want it to only run once take a look into [`getStaticProps`](https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation) instead, which will give you very fast client side navigations and static content instead of serverless functions.
